src/server/editorServices.ts

```diff
@@ -1105,187 +1312,159 @@ namespace ts.server {
+            this.logger.info(`Opened configuration file ${configFileName}`);
+            return this.createAndAddConfiguredProject(configFileName, projectOptions, configFileErrors, configFileSpecs, cachedServerHost, clientFileName);
         }
 
-        private updateNonInferredProject<T>(project: ExternalProject | ConfiguredProject, newUncheckedFiles: T[], propertyReader: FilePropertyReader<T>, newOptions: CompilerOptions, newTypeAcquisition: TypeAcquisition, compileOnSave: boolean, configFileErrors: Diagnostic[]) {
-            const oldRootScriptInfos = project.getRootScriptInfos();
-            const newRootScriptInfos: ScriptInfo[] = [];
-            const newRootScriptInfoMap: NormalizedPathMap<ScriptInfo> = createNormalizedPathMap<ScriptInfo>();
+        private updateNonInferredProjectFiles<T>(project: ExternalProject | ConfiguredProject, newUncheckedFiles: T[], propertyReader: FilePropertyReader<T>, clientFileName?: string) {
+            const projectRootFilesMap = project.getRootFilesMap();
+            const newRootScriptInfoMap: Map<ProjectRoot> = createMap<ProjectRoot>();
 
-            let projectErrors: Diagnostic[];
-            let rootFilesChanged = false;
             for (const f of newUncheckedFiles) {
                 const newRootFile = propertyReader.getFileName(f);
-                if (!this.host.fileExists(newRootFile)) {
-                    (projectErrors || (projectErrors = [])).push(createFileNotFoundDiagnostic(newRootFile));
-                    continue;
-                }
                 const normalizedPath = toNormalizedPath(newRootFile);
-                let scriptInfo = this.getScriptInfoForNormalizedPath(normalizedPath);
-                if (!scriptInfo || !project.isRoot(scriptInfo)) {
-                    rootFilesChanged = true;
-                    if (!scriptInfo) {
-                        const scriptKind = propertyReader.getScriptKind(f);
-                        const hasMixedContent = propertyReader.hasMixedContent(f, this.hostConfiguration.extraFileExtensions);
-                        scriptInfo = this.getOrCreateScriptInfoForNormalizedPath(normalizedPath, /*openedByClient*/ false, /*fileContent*/ undefined, scriptKind, hasMixedContent);
-                    }
-                }
-                newRootScriptInfos.push(scriptInfo);
-                newRootScriptInfoMap.set(scriptInfo.fileName, scriptInfo);
-            }
-
-            if (rootFilesChanged || newRootScriptInfos.length !== oldRootScriptInfos.length) {
-                let toAdd: ScriptInfo[];
-                let toRemove: ScriptInfo[];
-                for (const oldFile of oldRootScriptInfos) {
-                    if (!newRootScriptInfoMap.contains(oldFile.fileName)) {
-                        (toRemove || (toRemove = [])).push(oldFile);
+                let scriptInfo: ScriptInfo | NormalizedPath;
+                let path: Path;
+                if (!project.lsHost.fileExists(newRootFile)) {

```

Comment:
We reach through to `project.lsHost.fileExists` instead of `this.host.fileExists`? Would they return different results?